process/sync/baseForkDetector.go

```diff
@@ -611,3 +613,69 @@ func (bfd *baseForkDetector) computeGenesisTimeFromHeader(headerHandler data.Hea
 	genesisTime := int64(headerHandler.GetTimeStamp() - headerHandler.GetRound()*uint64(bfd.rounder.TimeDuration().Seconds()))
 	return genesisTime
 }
+
+func (bfd *baseForkDetector) addHeader(
+	header data.HeaderHandler,
+	headerHash []byte,
+	state process.BlockHeaderState,
+	selfNotarizedHeaders []data.HeaderHandler,
+	selfNotarizedHeadersHashes [][]byte,
+	doJobOnBHProcessed func(data.HeaderHandler, []byte, []data.HeaderHandler, [][]byte),
+) error {
+
+	if check.IfNil(header) {
```

Comment:
suggestion to split this in 2 parts (just like in interceptors):
1. check block validity which might output an error that needs to be returned so the block won't be propagated
2. process received block that will make this function always return nil.

Reply:
Done